Criado Retorno Sicoob padrão CNAB240 - Segmento T e U

mauroagr opened the pull request:
- Foi criado o arquivo de retorno para o Banco SICOOB padrão CNAB240, segmento T e U
- Alguns ajustes de tabulação e espaçamento foram removidos

mauroagr committed:
Criado Retorno Sicoob padrão CNAB240 - Segmento T e U

- Foi criado o arquivo de retorno para o Banco SICOOB padrão CNAB240, segmento T e U
- Alguns ajustes de tabulação e espaçamento foram removidos

manoelcampos commented on src/ManoelCampos/RetornoBoleto/RetornoCNAB240Sicoob.php:
```diff
@@ -0,0 +1,214 @@
+ * Layout Padrão <a href="http://www.febraban.org.br">Febraban</a> 240 posições V08.4 de 01/09/2009.<p/>
+ *
+ * @license <a href="https://opensource.org/licenses/MIT">MIT License</a>
+ * @author <a href="http://manoelcampos.com/contact">Manoel Campos da Silva Filho</a>
```
Coloque seu nome como autor, afinal de contas foi vc quem fez esta classe.

manoelcampos commented on src/ManoelCampos/RetornoBoleto/RetornoCNAB240Sicoob.php:
```diff
@@ -0,0 +1,214 @@
+ * @author <a href="http://manoelcampos.com/contact">Manoel Campos da Silva Filho</a>
+ * @version 1.1
+ */
+class RetornoCNAB240Sicoob extends RetornoAbstract {
```
A classe deveria estender RetornoCNAB240.

manoelcampos commented on src/ManoelCampos/RetornoBoleto/RetornoCNAB240Sicoob.php:
```diff
@@ -0,0 +1,214 @@
+        return substr($linha, 8, 1);
+    }
+
+    public function getTotalCaracteresPorLinha() {
```
Estendendo RetornoCNAB240, alguns métodos como este não precisam ser redefinidos.

manoelcampos reviewed (changes requested):
Obrigado pela contribuição. Existem só minimos detalhes que gostaria que corrigisse antes de eu aprovar a PR.

mauroagr committed:
Ajustes Gerais

- Autor
- classe para extender
- remoção de funções já extendidas.

mauroagr commented:
Manoel, 
Alterações feitas.
abraço
Mauro

manoelcampos committed:
Update RetornoFactory.php

O método tentaInstanciarCnab240() não pode gerar uma exceção, pois se o arquivo não estiver no formato CNAB240, ele pode estar no formato CNAB400. Assim, o método tentaInstanciarObjetoLeituraArquivoRetorno() vai primeiro chamar tentaInstanciarCnab240(). Se este retornar NULL, ele chama o tentaInstanciarCnab400(). Caso este também retorne NULL, aí sim uma exceção é lançada.

manoelcampos reviewed (approved):
Feita apenas uma alteração no método `tentaInstanciarCnab240()`. Veja mensagem do commit para detalhes.

Obrigado pela contribuição.
